### PR TITLE
WT-11051 Fix newest start durable timestamp comparison in aggregate timestamp validation

### PR DESCRIPTION
**Pull request by raviprakashgiri29**

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/support/timestamp.c [hunk 1]
```diff
@@ -272,12 +272,14 @@ __wt_time_aggregate_validate(
     /*
      * In the case of missing timestamps, we assign the start point to the stop point and newest
      * start durable timestamp may be larger than newest stop timestamp. Check whether start and
-     * stop are equal first.
+     * stop are equal first and then check the newest start durable timestamp against newest stop
```
Comment (bvpvamsikrishna) [inline src/support/timestamp.c:275]: This is confusing, could you elaborate a bit more, the relation between various timestamps and why these comparisons are correct.

Comment (raviprakashgiri29) [reply src/support/timestamp.c:326]: @bvpvamsikrishna I tried to elaborate on the comments as discussed, but I couldn't break the conditions under each scenario as the conditions and scenarios are negated.

File: src/support/timestamp.c [hunk 2]
```diff
@@ -245,6 +245,57 @@ __wt_time_aggregate_validate(
 {
     char time_string[2][WT_TIME_STRING_SIZE];
 
+    /*
+     * The aggregated time window values that are tracked at the page level.
+     *    newest_start_durable_ts - The default value is WT_TS_NONE. It tracks the maximum durable
+     timestamp of all the inserts performed on a page.
```
Comment (bvpvamsikrishna) [inline src/support/timestamp.c:251]: replace "inserts" with "inserts, updates, or modify operations"

File: src/support/timestamp.c [hunk 3]
```diff
@@ -245,6 +245,57 @@ __wt_time_aggregate_validate(
...
+     *    newest_stop_durable_ts - The default value is WT_TS_NONE. It tracks the maximum durable
+     timestamp of all the removes performed on a page.
```
Comment (bvpvamsikrishna) [inline src/support/timestamp.c:253]: replace "removes" with " the delete operations"

File: src/support/timestamp.c [hunk 4]
```diff
@@ -245,6 +245,57 @@ __wt_time_aggregate_validate(
...
+     transaction id of a remove operation on a page. If there is no removal for a key, this value
+     will be WT_TXN_MAX.
+     *
+     *
+     * Three scenarios might happen at any point of time.
+     * Scenario 1 - No deletes on the page, only inserts and updates.
+     *    newest_stop_ts will be WT_TS_MAX since there is no removal of any key.
+     *    newest_start_durable_ts will be some valid value (not WT_TS_MAX or WT_TS_NONE)
+     *    newest_stop_durable_ts will be WT_TS_NONE
+     *    oldest_start_ts will be the minimum commit timestamp of any inserts performed on a page.
```
Comment (bvpvamsikrishna) [inline src/support/timestamp.c:270]: could we have both start together and stop after wards.

Comment (raviprakashgiri29) [reply src/support/timestamp.c:268]: Fixed in d0719a967c76f22f12ccd16c910bb1b8acd8522b

File: src/support/timestamp.c (hunk 3 above)
Comment (raviprakashgiri29) [reply src/support/timestamp.c]: Fixed in d0719a967c76f22f12ccd16c910bb1b8acd8522b

File: src/support/timestamp.c (hunk 2 above)
Comment (raviprakashgiri29) [reply src/support/timestamp.c]: Fixed in d0719a967c76f22f12ccd16c910bb1b8acd8522b

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm.
Please replace a "remove" with "delete"

**Pull request merged**